File: Model/ViewModel/OneUserOrderVM.cs
using System.Collections.Generic;
using targeteo.pl.Model.ViewModel;

namespace targeteo.pl.Model.ViewModel
{
    public class OneUserOrderVM
    {
        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string PostalCode { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public bool Checkbox { get; set; }
        public string Status { get; set; }
        public UserVM User { get; set; }
        public string AspNetUserId { get; set; }

        public SupplyMethodVm[] SupplyMethods { get; set; }
        public UserSupplyMethodVm SelectedSupplyMethod { get; set; }
        public int? UserSupplyMethodId { get; set; }

        public UserSupplyMethodVm UserSupplyMethod { get; set; }
        public string PaczkomatDestinationAddress { get; set; }
        //cause it has Quantity and Productvm
        public ICollection<OrderDetailVM> OrderDetails { get; set; }

        public string OrderSummaryId { get; set; }
        public OrderSummaryVm OrderSummary { get; set; }
    }
}

File: ClientApp/src/app/model/oneUserOrderVm.ts
import { CartItem } from "../services/shopping-cart.service";
import { UserSupplyMethodVm } from "./userSupplyMethodVm";
import { SupplyMethodVm } from "./supplyMethodVm";
import { orderDetailVm } from "./orderDetailVm";
import { User } from "oidc-client";

export class OneUserOrderVm {

    public Id: number;
    public FirstName: string;
    public LastName: string;
    public Address: string;
    public City: string;
    public State: string;
    public PostalCode: string;
    public Email: string;
    public Phone: string;
    public SupplyMethods: SupplyMethodVm[];
    public SelectedSupplyMethod: UserSupplyMethodVm;
    public PaczkomatDestinationAddress: string;
    public PaymentUrl: string;
    public OrderSummaryId: string;
    public User: any;
    public Status: string;
    public StatusCaption: string;
    public OrderDetails: orderDetailVm[];

    public Items: CartItem[];
    // public checkbox: boolean;
}
